### PR TITLE
feature(ui-ux): dvm <> evm portfolio

### PR DESCRIPTION
**Pull request by nattadex**
#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: shared/contexts/DomainProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+interface DomainContextI {
+  api: {
+    get: () => Promise<string | null>;
+    set: (domain: NonNullable<string>) => Promise<void>;
+  };
+}
+
+export function useDomain({ api }: DomainContextI): DomainLoader {
+  const defaultDomain = "DFI";
```
Comment (lykalabrada) [inline shared/contexts/DomainProvider.tsx:23]: Is it possible to make an enum for domain?

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/PortfolioCard.tsx [hunk 2]
```diff
@@ -158,12 +167,18 @@ function getEmptyScreenDetails(type?: ButtonGroupTabKey): {
     case ButtonGroupTabKey.AllTokens:
     default:
       return {
-        icon: EmptyPortfolioIcon,
+        icon: domain === "DFI" ? EmptyPortfolioIcon : EmptyEvmPortfolioIcon,
```
Comment (lykalabrada) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/components/PortfolioCard.tsx:170]: I'm guessing for EVM this empty portfolio should only be displayed when DFI for EVM balance === 0. But let's wait for confirmation

Comment (nattadex) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/components/PortfolioCard.tsx:170]: yess!! Will add an additional condition of `DFI for EVM balance === 0` once transfer domain is up!

File: mobile-app/app/components/DomainSwitch.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  return (
+    <View
+      style={tailwind("h-7 w-18 rounded-full flex-row justify-center mr-4", {
+        "bg-mono-light-v2-300": isLight,
+        "bg-mono-dark-v2-300": !isLight,
+      })}
+      testID={testID}
+    >
+      <ThemedTouchableOpacityV2
```
Comment (lykalabrada) [inline mobile-app/app/components/DomainSwitch.tsx:26]: hmm thinking if we should just do something like this to reduce `isEvmDomain` conditions and css show/hide
```
{domain === DomainType.DFI && (<ThemedTouchableOpacityV2 ....>....</ThemedTouchableOpacityV2>)
{domain === DomainType.EVM && (<ThemedTouchableOpacityV2 ....>....</ThemedTouchableOpacityV2>)
```

**Review by lykalabrada — CHANGES_REQUESTED** [review]

File: mobile-app/app/components/DomainSwitch.tsx [hunk 4]
```diff
@@ -23,90 +23,76 @@ export function DomainSwitch({ testID }: { testID: string }): JSX.Element {
...
           style={tailwind(
-            "rounded-full p-1 flex-row items-center",
-            {
+            "flex-1 flex-row justify-center rounded-full items-center p-0 absolute left-0"
+          )}
+        >
+          <View
+            style={tailwind("rounded-full p-1 flex-row items-center", {
               "bg-transparent": isEvmDomain,
               "bg-mono-light-v2-1000": isLight && !isEvmDomain,
```
Comment (lykalabrada) [inline mobile-app/app/components/DomainSwitch.tsx:40]: We can also remove all `!isEvmDomain` in this section 🙂

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/components/PortfolioCard.tsx [hunk 5]
```diff
@@ -38,10 +40,13 @@ export function PortfolioCard({
...
   const { hasFetchedToken } = useSelector((state: RootState) => state.wallet);
-
+  const { domain } = useDomainContext();
   // return empty portfolio if no DFI and other tokens
   if (isZeroBalance) {
-    const screenDetails = getEmptyScreenDetails(ButtonGroupTabKey.AllTokens);
+    const screenDetails = getEmptyScreenDetails(
+      ButtonGroupTabKey.AllTokens,
+      domain
+    );
```
[the remark this answers is not in the document]
Comment (lykalabrada) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/components/PortfolioCard.tsx:49]: We can revisit on a separate PR 👍

**Review by lykalabrada — APPROVED** [review]

**Review by pierregee — APPROVED** [review]

**Pull request merged**